packages/utils/src/utils/router.js

```diff
@@ -118,7 +118,7 @@ export const paths = {
       });
     },
     byClusterTask() {
-      return '/taskruns?labelSelector=tekton.dev%2Ftask%3D:taskName';
+      return '/taskruns?kind=clusterTask&labelSelector=tekton.dev%2Ftask%3D:taskName';
```

Comment:
This raises an interesting issue, the label on ClusterTasks should probably be `tekton.dev/clustertask` rather than `tekton.dev/task` so we can differentiate between runs of a Task and ClusterTask with the same name.

Edit: actually looks like this was changed recently in pipelines... https://github.com/tektoncd/pipeline/issues/2377
From now on labels are covered by the API deprecation policy.

@a-roberts fyi. I'll take another look at this tomorrow, not sure if it's included in 0.11.2 or if it'll be 0.12 before we get this

Reply:
Good find! We could add `tekton.dev/clusterTask` to TaskRuns created from ClusterTasks in our API... but I think that TaskRuns created as part of a PipelineRun would be unaffected until we get the version of pipeline which does this. 

Also, TaskRuns created from ClusterTasks before this change wouldn't be listed on the TaskRuns page if we started using this label as a filter.

The point of this workaround is just to get the fact that it's a ClusterTask to the `CreateTaskRun` modal... I'm open to any suggestions on a better way to do that, since putting it in the URL doesn't feel ideal.

Reply:
I looked into this some more and it's actually been updated so the old label can be removed over multiple releases as per our API deprecation policy. This means that on pipelines master, TaskRuns referencing a ClusterTask now get both labels applied. The `tekton.dev/task` label on these runs will be marked as deprecated in the next (?) release, and removed in a subsequent release.

We still need to figure out how we're going to handle this in the dashboard.

I propose we do something like this:
1. update the URL to use the new label for filtering (replacing the current one) from the ClusterTasks page
  i. this includes updating the TaskRuns container's `mapStateToProps` to replace the old label with the new one
2. use the presence of the new label in the URL to determine which `kind` to use for the Create UI
3. in the container's `fetchTaskRuns` detect the presence of the new label and make 2 calls to `props.fetchTaskRuns`, 1 for each of the labels
4. in the container's `mapStateToProps` update `taskRuns` to return the union of 2 calls to the selector, 1 for each of the labels

Once we update to a pipelines release that removes the old label we can just remove the redundant call from 3 and 4.

Open to other suggestions if there's a cleaner way to handle this with minimal impact.

Reply:
Made this adjustment. It might be easier to check just those changes by looking at this commit: https://github.com/JustinKuli/dashboard/commit/bafee1e75c886ed2d33262ae138d23971bcd719b

Let me know what you think. I had to adjust some other logic to get the `kind` to appear correctly... there might still be something buggy with how I'm handling that in props vs state.